bundles/core/org.eclipse.smarthome.core/src/main/java/org/eclipse/smarthome/core/items/ItemUtil.java

```diff
@@ -82,26 +81,18 @@ public static void assertValidItemName(String itemName) throws IllegalArgumentEx
     }
 
     public static State convertToAcceptedState(State state, Item item) {
-        if (state == null) {
-            LOGGER.error("A conversion of null was requested", new NullPointerException("state should not be null")); // NOPMD
-            return UnDefType.NULL;
+        BundleContext bundleContext = FrameworkUtil.getBundle(ItemUtil.class).getBundleContext();
```

Comment:
I can only think of this change when we do break the API. Keeping the method `static` while accessing an instance variable (`itemStateConverter`) will not work.

Reply:
what I can do is change calls to `ItemUtils#convertToAcceptedState` to `ItemStateCOnverter#convertToAccpetedState` in all places in the core runtime (there are only two). This will limit the expensive bundle & service reference access to calls outside of ESH.
wdyt?

Reply:
> Keeping the method static while accessing an instance variable (itemStateConverter) will not work. 

Yes, that's clear. The field would thus have to be static as well (which is not a recommended way to do, but it would be comparable to using FrameworkUtil here but with the benefit that the resolution is only done once.

Reply:
I did replace both occurrences of the static call with the DS instance of ItemStateConverter and deprecated the static method in ItemUtils.